test/instances/ezETH/EzETHHyperdrive.t.sol

```diff
@@ -0,0 +1,1159 @@
+        (, , uint256 totalPooledEtherBefore) = RESTAKE_MANAGER.calculateTVLs();
+        uint256 totalSharesBefore = EZETH.totalSupply();
+        AccountBalances memory bobBalancesBefore = getAccountBalances(bob);
+        AccountBalances memory hyperdriveBalancesBefore = getAccountBalances(
+            address(hyperdrive)
+        );
+
+        // Bob opens a long by depositing ETH.
+        basePaid = basePaid.normalizeToRange(
+            2 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            HyperdriveUtils.calculateMaxLong(hyperdrive)
+        );
+        openLong(bob, basePaid);
+
+        // Ensure that Lido's aggregates and the token balances were updated
+        // correctly during the trade.
+        verifyDeposit(
+            bob,
+            basePaid,
+            true,
+            totalPooledEtherBefore,
+            totalSharesBefore,
+            bobBalancesBefore,
+            hyperdriveBalancesBefore
+        );
+    }
+
+    function test_open_long_refunds() external {
+        vm.startPrank(bob);
+
+        // Ensure that Bob receives a refund on the excess ETH that he sent
+        // when opening a long with "asBase" set to true.
+        uint256 ethBalanceBefore = address(bob).balance;
+        hyperdrive.openLong{ value: 2e18 }(
+            1e18,
+            0,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+        assertEq(address(bob).balance, ethBalanceBefore - 1e18);
+
+        // Ensure that Bob receives a  refund when he opens a long with "asBase"
+        // set to false and sends ether to the contract.
+        ethBalanceBefore = address(bob).balance;
+        hyperdrive.openLong{ value: 0.5e18 }(
+            1e18,
+            0,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: false,
+                extraData: new bytes(0)
+            })
+        );
+        assertEq(address(bob).balance, ethBalanceBefore);
+    }
+
+    function test_open_long_with_ezeth(uint256 basePaid) external {
+        // Get some balance information before the deposit.
+        (, , uint256 totalPooledEtherBefore) = RESTAKE_MANAGER.calculateTVLs();
+        uint256 totalSharesBefore = EZETH.totalSupply();
+        AccountBalances memory bobBalancesBefore = getAccountBalances(bob);
+        AccountBalances memory hyperdriveBalancesBefore = getAccountBalances(
+            address(hyperdrive)
+        );
+
+        // Bob opens a long by depositing stETH.
+        basePaid = basePaid.normalizeToRange(
+            2 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            HyperdriveUtils.calculateMaxLong(hyperdrive)
+        );
+
+        (, , uint256 totalPooledEther) = RESTAKE_MANAGER.calculateTVLs();
+        uint256 totalShares = EZETH.totalSupply();
```

Comment:
I would create a util function to go from base -> shares in the test file

Reply:
This was a nice cleanup, thanks.